### PR TITLE
Dashboard Projects Controller

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._
This PR adds in a Dashboard Projects controller with an index action, that replaces the index action from Projects controller.

The Dashboard Projects index page has searching and sorting powered by [ransack](https://github.com/activerecord-hackery/ransack). Projects can be searched by name and sorted on a number of attributes.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as your preferred user
2. Click on the `Projects` link on the sidebar or from the navbar
3. Click on `Personal` to view projects which belong to your users namespace
4. Type into the search box (labelled `Filter by name`) to reduce the listing of projects to ones which names contain your search phrase
5. Click on the `Updated At` dropdown and click on your preferred sort

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by joshsadam — COMMENTED** [review]
This looks really good.  There are the few things we discussed with the full page reloads, but as per our discussion, I know that they cannot be fixed for now.  As far as I can tell all works and looks great, just one minor comment below.

File: app/views/dashboard/projects/index.turbo_stream.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <tbody
+      class="
+        bg-white
+        divide-y
+        divide-gray-200
+        dark:bg-gray-800
+        dark:divide-gray-700
+      "
+    >
+      <%= render partial: "dashboard/projects/project", collection: @projects %>
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/views/dashboard/projects/index.turbo_stream.erb:12]: This does not make sense afer further discussion

File: app/controllers/dashboard/projects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Dashboard
+  # Dashboard Projects Controller
+  class ProjectsController < ApplicationController
+    def index # rubocop:disable Metric/AbcSize
+      @q = Project.ransack(params[:q])
```
Comment (deepsidhu85) [inline app/controllers/dashboard/projects_controller.rb:7]: Within this file and from wherever this is called can you please to update the variable and parameter `q` to something more descriptive

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking great. Just have the one comment below.

File: app/controllers/dashboard/projects_controller.rb (hunk 2 above)
Comment (joshsadam) [reply app/controllers/dashboard/projects_controller.rb:7]: `q` is kind of the standard for a query param like this I think

Comment (ericenns) [reply app/controllers/dashboard/projects_controller.rb:7]: `@q` is the standard value that represents the ransacked object. See documentation here https://activerecord-hackery.github.io/ransack/getting-started/simple-mode/

Comment (deepsidhu85) [reply app/controllers/dashboard/projects_controller.rb:7]: Thanks! @joshsadam @ericenns

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**